microsoft-edge/devtools-guide-chromium/memory-problems/heap-snapshots.md

```diff
@@ -263,47 +235,54 @@ To find an object in the collected heap, you can search using **Ctrl+F** and giv
 <!-- ====================================================================== -->
 ## Uncover DOM leaks
 
-The heap profiler has the ability to reflect bidirectional dependencies between browser native objects (DOM nodes, CSS rules) and JavaScript objects.  This helps to discover otherwise invisible leaks happening due to forgotten detached DOM subtrees floating around.
+The heap profiler has the ability to reflect bidirectional dependencies between browser native objects (DOM nodes, CSS rules) and JavaScript objects.  This helps to discover otherwise invisible leaks that happen because of forgotten detached DOM subtrees that remain in memory.
 
-DOM leaks may be bigger than you think.  Consider the following sample.  When is the `#tree` garbage-collected?
+Consider the following code sample.  When is the element with ID `tree` garbage-collected?
 
 ```javascript
+// Getting a reference to the #tree element.
 var treeRef = document.querySelector("#tree");
+
+// Getting a reference to the #leaf element,
+// which is a descendant of the #tree element.
 var leafRef = document.querySelector("#leaf");
-var body = document.querySelector("body");
 
-body.removeChild(treeRef);
+// Removing the #tree element from the DOM.
+document.body.removeChild(treeRef);
+
+// The #tree element can't be garbage-collected yet
+// because the treeRef variable still exists.
 
-//#tree in not GC yet due to treeRef
+// Removing the treeRef variable.
 treeRef = null;
 
-//#tree is not GC yet due to indirect reference from leafRef
+// The #tree element is still not garbage-collected
+// because of the indirect reference from the leafRef variable.
 
+// Removing the leafRef variable.
 leafRef = null;
-//#NOW can be #tree GC
+
+// Now the #tree element can be garbage-collected.
 ```
 
-The `#leaf` maintains a reference to the relevant parent (parentNode) and recursively up to `#tree`, so only when `leafRef` is nullified is the WHOLE tree under `#tree` a candidate for garbage-collection (GC).
+The element with ID `leaf` maintains a reference to its ancestor nodes in the DOM, which includes the element with ID `tree`. Both `treeRef` and `leafRef` must be nullified for the whole DOM tree under the element with ID `tree` to be garbage-collected.
 
 ![DOM subtrees](./heap-snapshots-images/memory-problems-tree-gc.png)
 
-
-### Demo webpage: Example 6: Leaking DOM nodes
+<!-- ------------------------------ -->
+#### Demo webpage: Example 6: Leaking DOM nodes
 
 Open the example webpage [Example 6: Leaking DOM nodes](https://microsoftedge.github.io/Demos/devtools-memory-heap-snapshot/example-06.html) in a new window or tab, to understand where DOM nodes might leak, and how to detect such leakage.
```

Comment:
```suggestion
To understand where DOM nodes might leak, and how to detect such leakage, open the example webpage [Example 6: Leaking DOM nodes](https://microsoftedge.github.io/Demos/devtools-memory-heap-snapshot/example-06.html) in a new window or tab.
```
antipattern (b/c reversed): 
Do A, B, and then C, if you want to D.
not always bad, but consider here.